src/commonMain/kotlin/fr/acinq/eclair/channel/Helpers.kt

```diff
@@ -357,6 +357,46 @@ object Helpers {
 
         fun isValidFinalScriptPubkey(scriptPubKey: ByteVector): Boolean = isValidFinalScriptPubkey(scriptPubKey.toByteArray())
 
+        fun btcAddressFromScriptPubKey(scriptPubKey: ByteVector, chainHash: ByteVector32): String? {
```

Comment:
It will be slightly more painful because it's a different lib, but I think this should be done in https://github.com/acinq/bitcoin-kmp

It's generally useful to have for any bitcoin application IMHO.

Reply:
The only argument I have against this is that, this code is lightning-specific. There's a related function `isValidFinalScriptPubkey`, which checks the script against the restrictions defined in Bolt 02. (Only 4 script formats allowed.) The new function `btcAddressFromScriptPubKey` is basically a repurpose of that function, which takes advantage of the limited script formats.

Also, as you mentioned below:

> You will be able to use `ClosingTx.toLocalOutput` to get the closing address once [#229](https://github.com/ACINQ/eclair-kmp/pull/229) lands

In which case this function would be removed.

Reply:
> The only argument I have against this is that, this code is lightning-specific.

I don't think it is, it covers the most used bitcoin script types. If some application using this translation from bitcoin-kmp needs another script type covered at some point, then it can be added to bitcoin-kmp and it benefits everyone. Since this PR will not land soon (IIUC we're freezing eclair-kmp soon for a release) I think it's worth taking this time to put it in bitcoin-kmp?

> In which case this function would be removed.

It won't remove this function, the `ClosingTx.toLocalOutput` will still give you a `scriptPubKey` that you'll need to translate to a bitcoin address. But it will make it a bit easier for you to get the info than matching the output against a stored `finalScriptPubKey`.